mslice/workspace/workspace.py

```diff
@@ -0,0 +1,103 @@
+import numpy as np
+import operator
+from base import WorkspaceBase
+from mantid.simpleapi import CreateWorkspace
+
+
+class Workspace(WorkspaceBase):
+    """workspace wrapper for MatrixWorkspace"""
+
+    def __init__(self, mantid_workspace):
+        self._raw_ws = mantid_workspace
```

Comment:
Thinking about this I wonder if we might be better with a `mixin-style` hierarchy (https://www.ianlewis.org/en/mixins-and-python). This would allows us to decouple the types in the inheritance hierarchy so that, for example, `PixelWorkspace` and `Workspace` do not inherit each other but both inherit `WorkspaceBase` and then we mixin other classes to add the functionality.